### ISSUE TITLE
Report.Prepare throws exception

### ISSUE THREAD

**Issue by lloydchriswilson** [FastReports/FastReport#562]
I am using open source version and trying to create a simple report (.fpx) from a template I created using the community designer (.frx). I want to be able to create an frx file using the designer but launch that report from my program substituting data having the same schema but filtered by my user. My frx file just connects to a sql server instance and displays selected fields from a table using a SQL query like (select a,b,c from table1). The preview in the designer looks fine.

Then in my VB.Net program (FR installed using latest version in Nuget) I want to convert the frx to fpx so that I can display it using the community viewer. My pseudocode looks like this:

```
dim report as new fastreport.report

report.load(frxname)

(create datatable dt containing identical data as used in frx)

report.registerdata(dt,"MyTable")

if report.prepare then report.saveprepared(fpxname)
```

At the report.prepare statement I get this error: 

System.TypeLoadException: 'Method 'GetLineMappings' in type 'ParsedSyntaxTree' from assembly 'Microsoft.CodeAnalysis.CSharp, Version=3.3.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' does not have an implementation.'

I suspect that the Nuget package version has incompatible referenced libraries. I tried to roll back to older versions but got different errors each time.

I am using the latest version 2023.1.13 installed using Nuget.

Thanks in advance,

Chris

**Comment by MaratAlaev** [comment]
Hello!

Have you installed the nuget package 
which is responsible for connecting FastReport to your SQL?

Best regards 
Marat

**Comment by lloydchriswilson** [comment]
Marat:

Thanks for replying to my question! I really appreciate it!

I did not realize that additional Nuget packages were needed depending on your provider. That’s actually a great way to do it.

I installed the MSSQL package and tried again but still get the same message:

System.TypeLoadException
  HResult=0x80131522
  Message=Method 'GetLineMappings' in type 'ParsedSyntaxTree' from assembly 'Microsoft.CodeAnalysis.CSharp, Version=3.3.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' does not have an implementation.
  Source=Microsoft.CodeAnalysis.CSharp
  StackTrace:
   at Microsoft.CodeAnalysis.CSharp.CSharpSyntaxTree.ParseText(SourceText text, CSharpParseOptions options, String path, ImmutableDictionary`2 diagnosticOptions, CancellationToken cancellationToken)
   at FastReport.Code.CSharp.CSharpCodeProvider.ParseTree(String text, CancellationToken ct)
   at FastReport.Code.CodeDom.Compiler.CodeDomProvider.CompileAssemblyFromSource(CompilerParameters cp, String code, CultureInfo cultureInfo)
   at FastReport.Code.AssemblyDescriptor.TryInternalCompile(CompilerParameters cp, CompilerResults& cr)
   at FastReport.Code.AssemblyDescriptor.InternalCompile()
   at FastReport.Code.AssemblyDescriptor.Compile()
   at FastReport.Report.Compile()
   at FastReport.Report.Prepare(Boolean append)
   at WRDB.frmMain.SavedFastReportClicked(Object sender, EventArgs e) in C:\Projects\WRDB 7.0\WRDB\Forms-Project\frmMain.vb:line 1725

I do not see the CodeAnalysis.CSharp is listed as a dependency. Do I need a different version installed?

UPDATE: I installed CodeAnalysis.CSharp 4.6.0-1.final and I no longer get this error.

Thanks for getting me going again,

Chris
